Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp

```diff
@@ -113,4 +113,10 @@ void WebRemoteFrameClient::focus()
         page->send(Messages::WebPageProxy::FocusRemoteFrame(m_frame->frameID()));
 }
 
+void WebRemoteFrameClient::dispatchDecidePolicyForNavigationAction(const NavigationAction& navigationAction, const ResourceRequest& request, const ResourceResponse& redirectResponse,
+    FormState* formState, const String& clientRedirectSourceForHistory, uint64_t navigationID, std::optional<WebCore::HitTestResult>&& hitTestResult, bool hasOpener, WebCore::SandboxFlags sandboxFlags, PolicyDecisionMode policyDecisionMode, FramePolicyFunction&& function)
+{
+    WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction(navigationAction, request, redirectResponse, formState, clientRedirectSourceForHistory, navigationID, WTFMove(hitTestResult), hasOpener, sandboxFlags, policyDecisionMode, WTFMove(function));
```

Comment:
This function looks the same as `WebLocalFrameLoaderClient::dispatchDecidePolicyForNavigationAction()`. Can't we just remove both of these functions and make `WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction()` final?

Reply:
That doesn't work because of multiple inheritance weirdness.  They inherit the need for dispatchDecidePolicyForNavigationAction through different ancestors.